models/experimental.py

```diff
@@ -86,6 +88,39 @@ def forward(self, x, augment=False, profile=False, visualize=False):
         y = torch.cat(y, 1)  # nms ensemble
         return y, None  # inference, train output
 
+class Custom_Layer(nn.Module):
+    # Custom layer for preprocessing
+    def __init__(self):
+        super().__init__()
+
+    def forward(self,input_tensor):
+        '''
+        **Resizing and Padding**
+        interpolation = T.InterpolationMode.NEAREST
+        transformer = torch.nn.Sequential(T.Pad((0,64)),T.Resize((640,640),interpolation=interpolation), antialias=False)
+        transformed_tensor = transformer(input_tensor)
+        Note: If using resize inside model, make sure the imgsz variable in export.py is changed accordingly to input image size
+        '''
+        transformed_tensor = input_tensor/255
+        transformed_tensor = torch.flip(transformed_tensor, [1])
+        return transformed_tensor
+
+# class which creates model with inbuilt preprocessing (defined in Custom_Layer)
+class Custom_Model(nn.Module):
+    def __init__(self, pretrained_model, nc, names, stride):
+        super(Custom_Model, self).__init__()
```

Comment:
You don't have to send the arguments nc, names and stride. You can access them from pretrained_model itself

pretrained_model.nc
pretrained_model.names
pretrained_model.stride